src/Microsoft.Azure.WebJobs.Host/Tables/TableEntityWriter.cs

```diff
@@ -172,23 +172,35 @@ private static ITableEntity Copy(ITableEntity item)
                 {
                     if (!e.IsNotFoundTableNotFound())
                     {
-                        throw;
+                        throw new StorageException(e.GetDetailedErrorMessage());
                     }
 
                     exception = e;
                 }
 
                 if (exception != null)
                 {
-                    // Make sure the table exists
-                    await _table.CreateIfNotExistsAsync(cancellationToken);
-
-                    // Commit the batch
-                    await _table.ExecuteBatchAsync(batch, cancellationToken);
+                    await CreateTableAndExecuteBatch(batch, cancellationToken);
                 }
             }
         }
 
+        private async Task CreateTableAndExecuteBatch(IStorageTableBatchOperation batch, CancellationToken cancellationToken)
+        {
+            // Make sure the table exists
+            await _table.CreateIfNotExistsAsync(cancellationToken);
+
+            // Commit the batch
+            try
+            {
+                await _table.ExecuteBatchAsync(batch, cancellationToken);
+            }
+            catch (StorageException e)
+            {
+                throw new StorageException(e.GetDetailedErrorMessage());
```

Comment:
@brettsam I wasn't sure if it would be more confusing to add the original exception as the inner exception, since I'm just adding more detail to the original (unhelpful) StorageException message.

`---> System.InvalidOperationException : Error while handling parameter outputTable after function returned: 
    ---> Microsoft.WindowsAzure.Storage.StorageException : Element 0 in the batch returned an unexpected response code. (HTTP status code 500: Table not found. Detailed message here)`

becomes

`---> System.InvalidOperationException : Error while handling parameter outputTable after function returned: 
    ---> Microsoft.WindowsAzure.Storage.StorageException : Element 0 in the batch returned an unexpected response code. (HTTP status code 500: Table not found. Detailed message here) ---> Microsoft.WindowsAzure.Storage.StorageException : Element 0 in the batch returned an unexpected response code. `

Is that ok?